### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by MozamelJawad**
**### Learning objectives**
- Understand the concept of state management.
- Use store, actions, and reducers in React.

In this step, I implement **Redux Toolkit** inside of a React application. Setup a Redux store and create a slice to save the state and reducers

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    addBook: (state, action) => {
+      const { title, author } = action.payload;
+      if (title || author) {
+        state.books.push({ ...action.payload });
+      }
+    },
+  },
+});
+
+// console.log(bookSlice);
```
Comment (Cabraham1) [inline src/redux/books/booksSlice.js]: - [ ] Nice work so far 👍🏿 please try not to leave commented codes on your code base

File: src/redux/books/booksSlice.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { createSlice } from '@reduxjs/toolkit';
+import books from '../../books';
+
+const initialState = {
+  books,
```
Comment (Cabraham1) [inline src/redux/books/booksSlice.js]: - [ ] Nice work so far 👍🏿  creating an initial book state but you book state is currently not holding a data type please attribute the book to an empty array.

 `book:[]`

File: src/redux/categories/categorySlice.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+};
+
+const categorySlice = createSlice({
+  name: 'category',
+  initialState,
+  reducers: {
+  },
+});
+
+export default categorySlice.reducer;
```
Comment (Cabraham1) [inline src/redux/categories/categorySlice.js]: - [ ] Nice work so far 👍🏿 but as per project requirement this is how you can correct this slice

 - Add Missing Initial State Field:
In the initial state of your code, add a field called design with the value 'Under Construction!'.

- Adjust Reducer Logic:
Modify the reducer logic inside your createSlice to update the design field based on the action payload. Here's the adjusted logic:

```
checkStatus: (state, action) => {
  state.design = action.payload === "Under construction" ? "Under Construction!" : state.design;
},
```
- Import and Export Correctly:
Make sure you are importing and exporting the createSlice function correctly from @reduxjs/toolkit.

- Export the Reducer:
At the end of your module, export the reducer using the .reducer property of the createSlice result, just like you did in the original code.

**Review by Cabraham1 — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @MozamelJawad  :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers, and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by usorfaitheloho — APPROVED** [review]
Hi @MozamelJawad  , 
# STATUS: APPROVED! 🥇 💯 

done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**